webapp/src/blocks.tsx

```diff
@@ -585,6 +587,35 @@ export class Editor extends toolboxeditor.ToolboxEditor {
                     return true
                 }
             });
+
+            const triggerEditorAction = (action: pxsim.SimulatorActionMessage["type"]) => {
+                switch (action) {
+                    case "escape": {
+                        this.parent.setSimulatorFullScreen(false);
+                        return;
+                    }
+                    case "navigateRegions" : {
+                        this.parent.showNavigateRegions();
+                        return
+                    }
+                }
+            }
+
+            const simulatorOrigins = [
+                window.location.origin,
+                // Simulator deployed origin.
+                "https://trg-microbit.userpxt.io"
+            ]
+            window.addEventListener("message", (e: MessageEvent<pxsim.SimulatorActionMessage>) => {
```

Comment:
This feels like it should be more positively identifying the message before making assumptions about its structure/type - let's compare to other sim messages and their handling.

Reply:
We now check the data type as `pxtsim` before trying to use the action [59d54e0](https://github.com/microsoft/pxt/pull/10551/commits/59d54e0c15e1dc158ab73634a35cb08c54f6e27b)